reco_utils/dataset/pandas_df_utils.py

```diff
@@ -74,3 +75,76 @@ def filter_by(df, filter_by_df, filter_by_cols):
             filter_by_df.set_index(filter_by_cols).index
         )
     ]
+
+
+def df_to_libffm(df, col_rating=DEFAULT_RATING_COL, filepath=None):
+    """Converts an input Dataframe (df) to another Dataframe (df) in libffm format. A text file of the converted
+    Dataframe is optionally generated.
+
+    Note:
+        The input dataframe is expected to represent the feature data in the following schema
+        |field-1|field-2|...|field-n|rating|
+        |feature-1-1|feature-2-1|...|feature-n-1|1|
+        |feature-1-2|feature-2-2|...|feature-n-2|0|
+        ...
+        |feature-1-i|feature-2-j|...|feature-n-k|0|
+        Where
+        1. each "field-*" occupies one column in the data, and
+        2. "feature-*-*" can be either a string or a numerical value.
+        3. If there are ordinal variables in the columns, users should make sure these columns
+        are properly converted to string type.
+
+        The above data will be converted to the libffm format by following the convention as explained in
+        https://www.csie.ntu.edu.tw/~r01922136/slides/ffm.pdf
+
+    Args:
+        df (pd.DataFrame): input Pandas dataframe.
+        col_rating (str): rating of the data.
+        filepath (str): path to save the converted data.
+
+    Return:
+        pd.DataFrame: data in libffm format.
+    """
+    df_new = df.copy()
+
+    # Check column types.
+    types = df_new.dtypes
+    if not all([x == object or x == np.integer or x == np.float for x in types]):
```

Comment:
per our discussion, let's use issubdtype for more generic type check. Other than this, everything looks good!